packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.h

```diff
@@ -81,6 +85,44 @@ typedef RCTBridge *_Nonnull (
  */
 @property (nonatomic, nullable) RCTCreateBridgeWithDelegateBlock createBridgeWithDelegate;
 
+/**
+ * Block that returns the location of the JavaScript source file. When running from the packager
+ * this should be an absolute URL, e.g. `http://localhost:8081/index.ios.bundle`.
+ * When running from a locally bundled JS file, this should be a `file://` url
+ * pointing to a path inside the app resources, e.g. `file://.../main.jsbundle`.
+ */
+@property (nonatomic, nullable) RCTSourceURLForBridgeBlock sourceURLForBridge;
+
+/**
+ * The bridge initializes any registered RCTBridgeModules automatically, however
+ * if you wish to instantiate your own module instances, you can return them
+ * from this block.
+ *
+ * Note: You should always return a new instance for each call, rather than
+ * returning the same instance each time the bridge is reloaded. Module instances
+ * should not be shared between bridges, and this may cause unexpected behavior.
+ *
+ * It is also possible to override standard modules with your own implementations
+ * by returning a class with the same `moduleName` from this method, but this is
+ * not recommended in most cases - if the module methods and behavior do not
+ * match exactly, it may lead to bugs or crashes.
+ */
+@property (nonatomic, nullable) RCTExtraModulesForBridgeBlock extraModulesForBridge;
+
+/**
+ * Retrieve the list of lazy-native-modules names for the given bridge.
+ */
+@property (nonatomic, nullable) RCTExtraLazyModuleClassesForBridge extraLazyModuleClassesForBridge;
+
+/**
+ * The bridge will call this block when a module been called from JS
+ * cannot be found among registered modules.
+ * It should return YES if the module with name 'moduleName' was registered
+ * in the implementation, and the system must attempt to look for it again among registered.
+ * If the module was not registered, return NO to prevent further searches.
+ */
+@property (nonatomic, nullable) RCTBridgeDidNotFindModuleBlock bridgeDidNotFindModule;
+
```

Comment:
nit: there're other optional functions in RCTBridgeDelegate
`shouldBridgeUseCustomJSC:`, `loadSourceForBridge:onProgress:onComplete:`, and `loadSourceForBridge:withBlock:`.
i don't think those are actually be used nowadays. just a note here that these functions are not going to support.
https://github.com/facebook/react-native/blob/1c52d385de662f034adf05df37073681cf609d78/packages/react-native/React/Base/RCTBridgeDelegate.h#L43-L74

Reply:
Looks like `shouldBridgeUseCustomJSC:` is not referenced at all in the codebase so Im not about it 😅 But I'll add the rest 

![CleanShot 2024-03-21 at 16 13 19@2x](https://github.com/facebook/react-native/assets/52801365/d3106b36-ebb7-4380-b0b6-17439bc84837)